### PR TITLE
disallow certain nginx configuraiton options

### PR DESCRIPTION
**Pull request by genisd**
do not allow placing (new) [edit: log] files outside of /data/
do not allow usage of client_body_temp_path

### CODE REVIEW

File: tests/test_assert_no_includes_in_config.py [hunk 1]
```diff
@@ -61,11 +62,68 @@ def test_include_prevention_illegal_includes(self):
...
+            "  access_log   /data/var/log/../../../access.log;",
+            "access_log   /tmp/staging.log;",
+            "access_log   output.log;",  # would be placed in /usr/share/nginx/output.log
+            "access_log '/var/log/nginx/acceptatie.log;'",
+            "     error_log \"/var/log/nginx/acceptatie.error.log info;\"",
+            "access_log   '/tmp/staging.log ';",
+            "access_log   \"output.log;\"",  # would be placed in /usr/share/nginx/output.log
+            "access_log  \"/usr/output.log;\"",
+            "access_log  '/tmp/staging.log ';",
+            "access_log  ' /tmp/staging.log';",
```
Comment (vdloo) [inline tests/test_assert_no_includes_in_config.py]: can we add a test for ../../../blabla ?

File: nginx_config_reloader/__init__.py [hunk 2]
```diff
@@ -147,6 +156,16 @@ def apply_new_config(self):
                     "https://support.hypernode.com/knowledgebase/how-to-use-nginx/\n"
                 )
                 return False
+
+        for rules in FORBIDDEN_CONFIG_REGEX:
+            try:
+                self.assert_regex_not_present(rules[0])
+            except subprocess.CalledProcessError:
+                error = "Unable to load config: " + rules[1]
```
Comment (vdloo) [inline nginx_config_reloader/__init__.py]: `error = "Unable to load config: {}".format(rules[1])`

File: nginx_config_reloader/__init__.py [hunk 3]
```diff
@@ -45,10 +45,20 @@
...
+# For security reasons the following nginx configuration parameters are forbidden
+FORBIDDEN_CONFIG_REGEX = \
+    [
+        ("client_body_temp_path", "Usage of configuration parameter client_body_temp_path is not allowed."),
+        ("^(?!\s*#)\s*(access|error)_log\s*"
+         "(\\042|\\047)?\s*"
+         "(?!(off|on|/+data/+))(?=.*\.\.|/+(?!data)|\w)"
+         "(\\042|\\047)?\s*",
+         "It's not allowed store access_log or error_log outside of /data/.")
+     ]
```
Comment (vdloo) [inline nginx_config_reloader/__init__.py]: shouldn't the ILLEGAL_INCLUDE_REGEX also be in the FORBIDDEN_CONFIG_REGEX list? also the \ is a bit strange here. why not just do 
```
FORBIDDEN_CONFIG_REGEX = [
    ...
]
```

Comment (genisd) [inline nginx_config_reloader/__init__.py]: It's to differentiate between the two errors (to get a different error message).
But indeed we should merge this, they seem similar. I thought of that too, let me think if there was any (good) reason not to do it.

Comment (genisd) [inline nginx_config_reloader/__init__.py]: this is the reason why it didn't seem feasible:
`if not self.allow_includes:`
https://github.com/ByteInternet/nginx_config_reloader/pull/12/files/57ed3ec650c1aed51f57681468f4dae74dfc4203#diff-19646a2bc74521520988231c63aa9f8bR147

I don't think this tool should allow out of place includes breaking the atomicity of applying new configs. If we ditch that option it'll save us a few lines of code.

Couldn't come up with a good implementation without removing that flag.
Should we remove it?

File: tests/test_assert_no_includes_in_config.py (hunk 1 above)
Comment (genisd) [inline tests/test_assert_no_includes_in_config.py]: done

**Comment by genisd** [comment]
should all be fixed now, can you take another look?

**Comment by vdloo** [comment]
:+1: could you create another PR that based these changes on the systemd-compat branch? I'd like to just deprecate the precise version and do all further developments on the xenial compatible version. I'll merge this into master for precise.

**Comment by genisd** [comment]
xenial based PR: https://github.com/ByteInternet/nginx_config_reloader/pull/13

**Pull request merged**